elastic/kibana
# [Reporting] Capture Kibana stopped error

## Summary

Partially addresses https://github.com/elastic/kibana/issues/125139

Introduces a new Reporting error code for representing when Kibana shuts down during report generation.

**Note**: While generating a PDF/PNG report we currently still capture the Chromium shutting down error. This is because running a reporting task fails due to Chromium unexpectedly closing long before it receives the message that Kibana is closing. Thus, this is an edge case error message for PNG/PDF reports.

## How to test

1. Start Kibana (enable the trial so that you have access to reporting)
2. Find Kibana's PID (I used `ps aux | grep 'kibana'`), in another terminal have `kill -15 <pid>` ready. This will soft-kill Kibana, allow it time to clean up and stop plugins gracefully.
3. Load sample data dashboard
4. Generate a PDF report
5. Wait for the log message `Compiling PDF using "preserve_layout" layout...`
6. Run the `kill` command from above
7. View the report output. You should see:

```json
...
"output": {
  "content_type": null,
  "warnings": [
    "Max attempts (1) reached for job l0go0y2l1dfj9d0062eorpid. Failed with: ReportingError(code: kibana_shutting_down_error)"
  ],
  "error_code": "kibana_shutting_down_error"
...
```




### Checklist
